txpipe/psf_diagnostics.py

```diff
@@ -272,9 +273,373 @@ def save_stats(self, rowe_stats):
+        print(f"Computing galaxy-cross-star statistic from {n} stars and {i} galaxies")
+        import treecorr
+        corr = treecorr.GGCorrelation(self.config)
+        cat1 = treecorr.Catalog(ra=ra, dec=dec, g1=q1[0], g2=q1[1], ra_units='deg', dec_units='deg')
+        cat2 = treecorr.Catalog(ra=ra, dec=dec, g1=q2[0], g2=q2[1], ra_units='deg', dec_units='deg')
+        corr.process(cat1, cat2)
+        return corr.meanr, corr.xip, corr.varxip**0.5
+
+    def galaxy_star_plots(self, galaxy_star_stats):
+        import matplotlib.pyplot as plt
+        import matplotlib.transforms as mtrans
+
+        f = self.open_output('star_shear_test', wrapper=True, figsize=(10, 6*len(STAR_TYPES)))
+        for s in STAR_TYPES:
+            ax = plt.subplot(len(STAR_TYPES), 1, s+1)
+            for j,i in enumerate([1,2]):
+                theta, xi, err = galaxy_star_stats[i, s]
+                tr = mtrans.offset_copy(ax.transData, f.file, 0.05*j-0.025, 0, units='inches')
+                if i==0:
+                    plt.errorbar(theta, abs(xi), err, fmt='.', label=rf'galaxy cross star', capsize=3, transform=tr)
+                else:
+                    plt.errorbar(theta, abs(xi), err, fmt='.', label=rf'galaxy cross star residuals', capsize=3, transform=tr)
+                plt.title(STAR_TYPE_NAMES[s])
+                plt.bar(0.0,2e-05,width=5,align='edge',color='gray',alpha=0.2)
+                plt.bar(5,1e-07,width=245,align='edge',color='gray',alpha=0.2)
+                plt.xscale('log')
+                plt.yscale('log')
+                plt.xlabel(r"$\theta$")
+                plt.ylabel(r"$\xi_+(\theta)$")
+                plt.legend()
+        f.close()
+
+    def save_stats(self, galaxy_star_stats):
 
+        f = self.open_output('star_shear_stats')
+        g = f.create_group("star_cross_galaxy")
+        #TODO add a more informative name than 1,2 for this type
+        for i in 1,2,3,4:
+            for s in STAR_TYPES:
+                theta, xi, err = galaxy_star_stats[i, s]
+                name = STAR_TYPE_NAMES[s]
+                h = g.create_group(f'star_cross_galaxy_{i}_{name}')
+                h.create_dataset('theta', data=theta)
+                h.create_dataset('xi_plus', data=xi)
+                h.create_dataset('xi_err', data=err)
+        f.close()
 
 
+
+
+class TXStarDensityTests(PipelineStage):
+    """
+    Star cross galaxy and star autocorrelation density stats.
+    """
+
+    name = 'TXGalaxyStarDensity'
+
+    inputs =[('shear_catalog', ShearCatalog),('star_catalog', HDFFile),
+    ('shear_tomography_catalog', TomographyCatalog),
+    ('random_cats', RandomsCatalog)]
+    outputs = [
+        ('star_density_test', PNGFile),
+        ('star_density_stats', HDFFile),
+    ]
+
+    config_options = {
+        'min_sep':0.5,
+        'max_sep':250.0,
+        'nbins':20,
+        'bin_slop':0.1,
+        'sep_units':'arcmin',
+        'psf_size_units':'sigma',
+        'shear_catalog_type':'metacal',
```

Comment:
Don't think that shear_catalog_type has to be here - it is added automatically later.